Sources/Kitura/String+Extensions.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright IBM Corporation 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import LoggerAPI
+
+extension String {
+    
+    /// Parses percent encoded string into query parameters
+    var asUrlEncoded: [String : String] {
```

Comment:
This is decoding, not encoding, and so should be called `urlDecoded`, or even better: `urlDecodedFieldValuePairs` to more clearly describe what it does.